Objects/StrategyCardAutomator.ttslua

```diff
@@ -968,27 +968,26 @@ function _addTechnologyPrimarySecondary(layout)
 
 	if not _setupHelper.getPoK() then
 		-- remove pok tech from ui
-
+		local _techSourceSet = _technologyHelper.getTechSourceSet()
 		local techlist = {["greenTechs"] = true, ["yellowTechs"] = true, ["redTechs"] = true, ["blueTechs"] = true}
-		local horizontalKeep = {}
 		for _,o in pairs(layout[technologyIndex]["children"]) do
 			local counter = 1
 			if o["tag"] == "HorizontalLayout" and techlist[o["attributes"]["id"]] then
-				for iv,v in ipairs(o["children"]) do
-					if not (iv == 2) and not (iv == 4) then
-						horizontalKeep[counter] = v
+				local horizontalKeep = {}
+				for ibutton,button in ipairs(o["children"]) do
+					local techName = button['attributes']['onClick']
+					source = _techSourceSet[string.sub(techName, 10, -2)]
+					if not (source == "PoK") then
+						horizontalKeep[counter] = button
```

Comment:
techName is identified by the name in the onClick function, namely "pickTech(techName)".
As getPoK is currently the only "find expansion" function exposed, its hard to make the whole thing apply for arbitrary expansions. At least now it should not break if positions change.

Reply:
Seems fine, it's clear what would need to change if/when the next expansion comes out.